contracts/modules/dispute/DisputeModule.sol

```diff
@@ -239,19 +260,69 @@ contract DisputeModule is
         emit DisputeCancelled(disputeId, data);
     }
 
+    /// @notice Tags a derivative if a parent has been tagged with an infringement tag
+    /// @param parentIpId The infringing parent ipId
+    /// @param derivativeIpId The derivative ipId
+    /// @param parentDisputeId The dispute id that tagged the parent ipId as infringing
+    function tagDerivativeIfParentInfringed(
+        address parentIpId,
+        address derivativeIpId,
+        uint256 parentDisputeId
+    ) external {
+        DisputeModuleStorage storage $ = _getDisputeModuleStorage();
+
+        Dispute memory parentDispute = $.disputes[parentDisputeId];
+        if (parentDispute.targetIpId != parentIpId) revert Errors.DisputeModule__ParentIpIdMismatch();
+        if (parentDispute.currentTag == IN_DISPUTE || parentDispute.currentTag == bytes32(0))
```

Comment:
you probably need to have an explicit check on the tag?

Reply:
Wasn't clear - what did you mean?

Reply:
you checked the currentTag == IN_DISPUTE, what's the value of the currentTag if the dispute is succeeded?  Do you need to check this condition in this function here?

Reply:
1. while in dispute - currentTag == IN_DISPUTE
2. if a dispute succeeds then the tag can be anything SP whitelisted - ie. "PLAGIARISM".
3. if a dispute fails then tag = address(0)

Derivatives can only be tagged in a permissionless way if the parent already has a tag like "PLAGIARISM"

Reply:
what's the default value of currentTag?

Reply:
do you want to add a helper function to check if the dispute is not succeeded yet?

Reply:
was double-checking just now - given this check is only used one - it would be strange to create an internal function just for it. Will add comments to make it more clear.

Reply:
https://github.com/storyprotocol/protocol-core-v1/pull/60/commits/3f73ef2cf048912644bbe8252e2d285900d3b2fd